### PR TITLE
nav2_controller: add loop rate log

### PR DESCRIPTION
**Pull request by ARK3r**
Following the format [here](https://github.com/ros-planning/navigation2/blob/c9d9b5c698d1f5aff060806481d18ba7e9b18cb2/nav2_planner/src/planner_server.cpp#L444) added loop rate logging to `nav2_controller/controller_server.cpp`.

### CODE REVIEW

**Review by SteveMacenski — APPROVED** [review]
No beef with the content of this PR, good contribution! Please fix the build and linting issues though before it can be merged.

File: nav2_controller/src/controller_server.cpp [hunk 1]
```diff
@@ -479,10 +481,13 @@ void ControllerServer::computeControl()
...
       }
 
+      auto cycle_duration = this->now() - start_time;
+
       if (!loop_rate.sleep()) {
         RCLCPP_WARN(
-          get_logger(), "Control loop missed its desired rate of %.4fHz",
-          controller_frequency_);
+          get_logger(),
+          "Control loop missed its desired rate of %.4f Hz.  Current loop rate is %.4f Hz",
```
Comment (SteveMacenski) [inline nav2_controller/src/controller_server.cpp:489]:
```suggestion
          "Control loop missed its desired rate of %.4f Hz. Current loop rate is %.4f Hz.",
```

File: nav2_controller/src/controller_server.cpp [hunk 2]
```diff
@@ -479,10 +481,13 @@ void ControllerServer::computeControl()
         break;
       }
 
+      auto cycle_duration = this->now() - start_time;
+
```
Comment (SteveMacenski) [inline nav2_controller/src/controller_server.cpp:485]:
```suggestion
```

File: nav2_controller/src/controller_server.cpp [hunk 3]
```diff
@@ -419,6 +419,8 @@ void ControllerServer::computeControl()
 {
   std::lock_guard<std::mutex> lock(dynamic_params_lock_);
 
+  auto start_time = this->now();
+
```
Comment (SteveMacenski) [inline nav2_controller/src/controller_server.cpp:423]:
```suggestion
```

**Comment by SteveMacenski** [comment]
@ARK3r this was reverted in https://github.com/ros-planning/navigation2/pull/4210 this fundamentally did not work. You set the start time to the start of the control cycle rather than at that current computational iteration.

In the future **please test** your PRs before submitting them.  This was a really silly mistake that could have been immediately caught if you ever attempted to use that log.

**Comment by ARK3r** [comment]
Hi @SteveMacenski, this was a rookie mistake and at the time I simply wasn't thinking it through. I'll be sure to not let this happen again.

**Comment by SteveMacenski** [comment]
If you want to resubmit another PR to add this feature (that works :wink: ) We're still open to that feature!

**Pull request merged**